src/main/java/edu/texas/social_computing/hospitals/RHRP.java

```diff
@@ -0,0 +1,134 @@
+package edu.texas.social_computing.hospitals;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+import java.io.FileNotFoundException;
+import java.util.*;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.Lists.newArrayList;
+
+public class RHRP {
+    private RHRP() {
+    }
+
+    /**
+     * Real HRP with Couples From http://www.nrmp.org/couples-match-videos/
+     * - Treat couples as a unit
+     * - Only make a match if the hospital is on resident's pref list and resident is on hospital's
+     * - If one partner has failed to match, increment the other's pref rank
+     * - If one partner has been kicked out due to being the worst, then break the other partner's match (it it has one)
+     * - Assumes that both partners have same pref list size
+     */
+
+    public static Matching run(HospitalTable hospitalTable,
+                               ResidentTable residentTable,
+                               Queue<Resident> freeResidents,
+                               Matching existingMatching) {
+        Matching m = existingMatching;
+
+        while (!freeResidents.isEmpty()) {
+            Resident currentResident = freeResidents.poll();
+            if (currentResident.getPreferences().isEmpty()) {
+                if (!currentResident.hasPartner()) {
+                    continue;
+                }
+                if (residentTable.getResidentById(currentResident.getPartnerId()).getPreferences().isEmpty()) {
+                    continue;
+                }
+                // resident and partner have to have the same pref list size
+                assert (false);
+            }
+
+            Hospital hospital = hospitalTable.getHospitalById(currentResident.getPreferences()
```

Comment:
naming suggestion: `hospital` -> `nextPreferredHospital`

also define `partnerNextPreferredHostpital` and use that in the partner `tryToMatch` below.